render.js

```diff
@@ -0,0 +1,361 @@
+            item.remove();
+            isCreateEnabled = true;
+        });
+
+        // appending the todo_button div to the unsaved cards div
+        node.appendChild(todo_buttons);
+
+        // Prepend the element to the DOM as the first child of the 
+        // element referenced by the 'todo_cards' element
+        todo_cards.prepend(node);
+   }
+   
+   else
+   {
+       console.log("Create is disabled");
+   }
+
+}
+
+export async function renderAllData(search_text)
+{
+    // selecting the todo card div containing all the cards
+    const todo_cards = document.querySelector('.todo-cards'); 
+    
+    // deleting all the cards
+    while(todo_cards.firstChild)
+    {
+        todo_cards.removeChild(todo_cards.firstChild);
+    }
+
+    // fetching the card data from the DB
+    const {error, data} = await getFromDB(search_text);
+
+    if(error)
+    {
+        throw new Error("Error while getting filtere data");
+    }
+
+    // setting initial as the minimum of data.length and limit
+    const initial = (data.length > limit) ? limit : data.length;
+
+    for (let i = 0; i < initial; ++i)
+    {
+        let todo = data[i];
+        createCard(todo);
+    }
+
+    // all button activated
+    const all_button = document.querySelector('#all-button');
+    if (all_button.disabled === true) all_button.disabled = false;
+
+    // load more button created if there are more cards than initial limit
+    if (initial < data.length)
+    {
+        const load_more = document.createElement("div");
+        load_more.setAttribute('class', 'load-more');
+
+
+        const load_more_button = document.createElement("button");
+        load_more_button.setAttribute('id', 'load-more-button');
+        load_more_button.appendChild(document.createTextNode("Load More"));
+    
+        load_more.appendChild(load_more_button);
+
+        todo_cards.appendChild(load_more);
+    
+        // load more button functionality
+        load_more_button.addEventListener('click', () => {
+            load_more_button.disabled = true;
+            load_more_button.style.display = 'none';
+    
+            for (let i = initial; i < data.length; ++i)
+            {
+                let todo = data[i];
+                createCard(todo);
+            }
+    
+        });
+    }
+
+}
+
+export async function renderFilteredData(type, search_text)
+{
+    // selecting the todo card div containing all the cards
+    const todo_cards = document.querySelector('.todo-cards'); 
+    
+    // deleting all the cards
+    while(todo_cards.firstChild)
+    {
+        todo_cards.removeChild(todo_cards.firstChild);
+    }
+
+    // fetching the card data from the DB based on the filter
+
+    // When the filter is incomplete
+    if (type === 'incomplete')
+        var {error, data} = await getIncompleteData(search_text);
+
+    else if (type === 'complete')
+        var {error, data} = await getCompleteData(search_text);
```

Comment:
We can merge this in one line to be like this
`const { error,  data } = await getDataWithCompletionStatus(searchText, completionType)`